pkg/internal/kubectl/execProvider.go

```diff
@@ -0,0 +1,32 @@
+/*Package kubectl xxx
+ */
+//  Old - mockgen -destination=pkg/internal/kubectl/mockExecProvider.go -package=kubectl -source=pkg/internal/kubectl/execProvider.go gitlab.fmr.com/common-platform/addons-manager/pkg/internal/kubectl ExecProvider
+//go:generate mockgen -destination=mockExecProvider.go -package=kubectl -source=execProvider.go . ExecProvider
+//go:generate mockgen -destination=../mocks/logr/mockLogger.go -package=mocks github.com/go-logr/logr Logger
+package kubectl
+
+import (
+	"os/exec"
+)
+
+// ExecProvider interface defines functions Kubectl uses to verify and execute a local command
+type ExecProvider interface {
+	findOnPath(file string) (string, error)
+	execCmd(name string, arg ...string) ([]byte, error)
+}
+
+// OsExecProvider implements the ExecProvider interface using the os/exec go module
+type OsExecProvider struct{}
```

Comment:
I usually embed the interface in the structure so it can be used as an interface, see https://github.com/fidelity/kraan/blob/master/pkg/internal/layers/layers.go#L30